tests/unit/s2n_alerts_test.c

```diff
@@ -51,6 +51,70 @@ int main(int argc, char **argv)
             EXPECT_SUCCESS(s2n_connection_free(conn));
             EXPECT_SUCCESS(s2n_disable_tls13());
         }
+
+        /* Test warning behavior */
+        {
+            const uint8_t warning_alert[] = { 0x01 /* AlertLevel = warning */,
+                                              0x46 /* AlertDescription = protocol_version (arbitrary value) */};
+
+            /* Warnings treated as errors by default */
+            {
+                struct s2n_connection *conn;
+                EXPECT_NOT_NULL(conn = s2n_connection_new(S2N_CLIENT));
+                EXPECT_EQUAL(conn->config->alert_behavior, S2N_ALERT_FAIL_ON_WARNINGS);
+                EXPECT_EQUAL(s2n_connection_get_protocol_version(conn), S2N_TLS12);
+
+                EXPECT_SUCCESS(s2n_stuffer_write_bytes(&conn->in, warning_alert, sizeof(warning_alert)));
+
+                EXPECT_FAILURE_WITH_ERRNO(s2n_process_alert_fragment(conn), S2N_ERR_ALERT);
+                EXPECT_TRUE(conn->closed);
+
+                EXPECT_SUCCESS(s2n_connection_free(conn));
+            }
+
+            /* Warnings ignored in TLS1.2 if alert_behavior == S2N_ALERT_IGNORE_WARNINGS */
```

Comment:
Is this the case for "close_notify" alerts as well? How is `close_notify` alerts treated in TLS1.2 and TLS1.3?

Reply:
"close_notify" is treated the same in TLS1.2 and TLS1.3. "close_notify" is not a warning and always closes the connection: [code](https://github.com/awslabs/s2n/blob/ba14c3b1378673954209792c4d0890f847846472/tls/s2n_alerts.c#L99-L103)